src/HookDispatcher.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * 2007-2019 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShop\Module\FacetedSearch;
+
+use Ps_Facetedsearch;
+use PrestaShop\Module\FacetedSearch\Hook;
+
+class HookDispatcher
```

Comment:
I like this system which allows to have a cleaner module class, but I'm afraid `HookDispatcher` might be confusing as it already exists in the core (we have an interface, an adapter service, a legacy hook dispatcher, ...) This name is already used quite a few times and for another purpose so maybe we can find something else for this module, although I don't have a clear idea of what it should be called.. `HookCatalog`, `HookManager`, `HookExecuter`, `HookStore`, ... Wdyt?

Reply:
If we must take in consideration Core classe, we can't have Category, Hook, Attribute, etc. classes.
This classe is here to dispatch hook, so we must don't care about Core name and called it as it should.

Reply:
And this is why we have namespaces :smile:

Reply:
Yeah I know we have namespaces... alright fair point for you!

Reply:
\o/ Beer?